src/odemis/gui/cont/tabs.py

```diff
@@ -2429,36 +2477,69 @@ def _enable_movement_controls(self, cancelled=False):
         Enable/disable chamber move controls (position and stage) based on current move
         :param cancelled: (bool) if the move is cancelled
         """
-        stage = self.tab_data_model.main.stage
-        # Get current movement (including unknown and on the path)
-        self._current_position = getCurrentPositionLabel(stage.position.value, stage)
-        self._enable_position_controls(self._current_position, cancelled)
-        # Enable stage advanced controls on milling
-        self._enable_advanced_controls(True) if self._current_position is MILLING else self._enable_advanced_controls(
-            False)
+        if self.tab_data_model.main.role == 'enzel':
+            stage = self.tab_data_model.main.stage
+            # Get current movement (including unknown and on the path)
+            self._current_position = getCurrentPositionLabel(stage.position.value)
+            # Enable stage advanced controls on milling
+            self._enable_advanced_controls(True) if self._current_position is MILLING else self._enable_advanced_controls(
+                False)
+
+        elif self.tab_data_model.main.role == 'meteor':
+            self._current_position = getCurrentPositionLabel(model.getComponent(role='stage-bare').position.value)
+        self._enable_position_controls(self._current_position, cancelled)            
```

Comment:
If you have self.stage as attribute, this can be rewritten as:

```
# Get current movement (would work for both enzel and metero)
self._current_position = getCurrentPositionLabel(stage.position.value)
self._enable_position_controls(self._current_position, cancelled)

# Enable stage advanced controls (you could put the enzel check inside this function to return immediately if it's not enzel)
 self._enable_advanced_controls(True) if self._current_position is MILLING else self._enable_advanced_controls(
                False)
```

Reply:
For the first part, okay. 
But it would be really odd to put the enzel check inside the function `self._enable_advanced_controls()`. If this function is made solely for enzel, why would i check inside it if the model is enzel or meteor?